libs/coin-modules/coin-evm/src/__tests__/coin-tester/indexer.ts

```diff
@@ -0,0 +1,220 @@
+        from: tx.from,
+        to: tx.to!,
+        transfer_events: isERC20
+          ? [
+              {
+                contract: log.address,
+                count: ethers.BigNumber.from(log.data).toString(),
+                from: ethers.utils.defaultAbiCoder.decode(["address"], log.topics[1])[0],
+                to: ethers.utils.defaultAbiCoder.decode(["address"], log.topics[2])[0],
+              },
+            ]
+          : [],
+        erc721_transfer_events: isERC721
+          ? [
+              {
+                contract: log.address,
+                sender: ethers.utils.defaultAbiCoder.decode(["address"], log.topics[1])[0],
+                receiver: ethers.utils.defaultAbiCoder.decode(["address"], log.topics[2])[0],
+                token_id: ethers.BigNumber.from(log.topics[3]).toString(),
+              },
+            ]
+          : [],
+        erc1155_transfer_events: isERC1155
+          ? [
+              {
+                contract: log.address,
+                sender: ethers.utils.defaultAbiCoder.decode(["address"], log.topics[2])[0],
+                receiver: ethers.utils.defaultAbiCoder.decode(["address"], log.topics[3])[0],
+                operator: ethers.utils.defaultAbiCoder.decode(["address"], log.topics[1])[0],
+                transfers: [
+                  Object.fromEntries(
+                    ethers.utils.defaultAbiCoder
+                      .decode(["uint256", "uint256"], log.data)
+                      .map((value, index) => [index === 0 ? "id" : "value", value.toString()]),
+                  ) as { id: string; value: string },
+                ],
+              },
+            ]
+          : [],
+        approval_events: [],
+        actions: [],
+        confirmations: tx.confirmations,
+        input: null,
+        gas_used: receipt.gasUsed.toString(),
+        cumulative_gas_used: receipt.cumulativeGasUsed.toString(),
+        status: receipt.status!,
+        received_at: new Date(block.timestamp * 1000).toISOString(),
+        block: {
+          hash: log.blockHash,
+          height: log.blockNumber,
+          time: new Date(block.timestamp * 1000).toISOString(),
+        },
+      };
+
+      const from = ethers.utils.defaultAbiCoder.decode(
+        ["address"],
+        isERC20 || isERC721 ? log.topics[1] : log.topics[2],
+      )[0];
+      const to = ethers.utils.defaultAbiCoder.decode(
+        ["address"],
+        isERC20 || isERC721 ? log.topics[2] : log.topics[3],
+      )[0];
+
+      if (!explorerAppendixByAddress[from]) {
+        explorerAppendixByAddress[from] = new Map();
+      }
+      explorerAppendixByAddress[from].set(operation.hash, operation);
+
+      if (!explorerAppendixByAddress[to]) {
+        explorerAppendixByAddress[to] = new Map();
+      }
+      explorerAppendixByAddress[to].set(operation.hash, operation);
+    }
+  }
+
+  return logs;
+};
+
+export const indexBlocks = async (): Promise<void> => {
```

Comment:
I don't see where this function is used 🤔

Reply:
Yes not used *yet* but let's keep it as it'll be useful soon